src/Controllers/BaseController.php

```diff
@@ -5,20 +5,37 @@
 namespace Mathematicator\Engine\Controller;
 
 
+use function get_class;
 use Mathematicator\Engine\Box;
 use Mathematicator\Engine\Context;
 use Mathematicator\Engine\DynamicConfiguration;
 use Mathematicator\Engine\Query;
 use Mathematicator\Engine\Source;
 use Mathematicator\Engine\TerminateException;
+use RuntimeException;
+use Symfony\Component\Translation\Translator;
+use Throwable;
 
 abstract class BaseController implements Controller
 {
 
+	/** @var Translator */
+	public $translator;
+
 	/** @var Context */
 	private $context;
 
 
+	/**
+	 * @param Translator $translator
+	 */
+	public function __construct(
```

Comment:
Please use `@inject` in property annotation instead of constructor. It is better for future support.

Reply:
fixed.